### ISSUE TITLE
Add GLC_FCS30D dataset

### ISSUE THREAD

**Issue by aazuspan** [aazuspan/sankee#52]
https://gee-community-catalog.org/projects/glc_fcs/

**Comment by aazuspan** [comment]
Need to verify, but this was crosswalked from a picture of the legend table by Gemini 🤯 

```python
labels = {
    10: "Rainfed cropland",
    11: "Herbaceous cover cropland",
    12: "Tree or shrub cover (Orchard) cropland",
    20: "Irrigated cropland",
    51: "Open evergreen broadleaved forest",
    52: "Closed evergreen broadleaved forest",
    61: "Open deciduous broadleaved forest (0.15<fc<0.4)",
    62: "Closed deciduous broadleaved forest (fc>0.4)",
    71: "Open evergreen needle-leaved forest (0.15< fc <0.4)",
    72: "Closed evergreen needle-leaved forest (fc >0.4)",
    81: "Open deciduous needle-leaved forest (0.15< fc <0.4)",
    82: "Closed deciduous needle-leaved forest (fc>0.4)",
    91: "Open mixed leaf forest (broadleaved and needle-leaved)",
    92: "Closed mixed leaf forest (broadleaved and needle-leaved)",
    120: "Shrubland",
    121: "Evergreen shrubland",
    122: "Deciduous shrubland",
    130: "Grassland",
    140: "Lichens and mosses",
    150: "Sparse vegetation (fc<0.15)",
    152: "Sparse shrubland (fc<0.15)",
    153: "Sparse herbaceous (fc<0.15)",
    181: "Swamp",
    182: "Marsh",
    183: "Flooded flat",
    184: "Saline",
    185: "Mangrove",
    186: "Salt marsh",
    187: "Tidal flat",
    190: "Impervious surfaces",
    200: "Bare areas",
    201: "Consolidated bare areas",
    202: "Unconsolidated bare areas",
    210: "Water body",
    220: "Permanent ice and snow"
}

palette = {
    10: "#FFFF64",
    11: "#FFFF64",
    12: "#FFFF00",
    20: "#AACCFF",
    51: "#4C7300",
    52: "#006400",
    61: "#AA8C00",
    62: "#00A000",
    71: "#005000",
    72: "#003C00",
    81: "#286400",
    82: "#285000",
    91: "#A0B432",
    92: "#788400",
    120: "#966400",
    121: "#964B00",
    122: "#966400",
    130: "#FFB432",
    140: "#D3CC9A",
    150: "#F2E2C7",
    152: "#D3A98A",
    153: "#F2E2C7",
    181: "#A8A870",
    182: "#70D8A2",
    183: "#9DB3CE",
    184: "#828282",
    185: "#F57EAA",
    186: "#64C8A5",
    187: "#44508D",
    190: "#C31200",
    200: "#F5F5DC",
    201: "#DDDDDD",
    202: "#F5F5DC",
    210: "#4285F4",
    220: "#FFFFFF"
}
```

**Comment by aazuspan** [comment]
After a closer look at the dataset, I'm going to hold off unless this is specifically requested.

**Comment by aazuspan** [comment]
Thanks @cchowgule, a PR would be welcome! I'm happy to go into more depth, but here are some initial thoughts on how to implement this.

Since GLC_FCS30D stores years in an unconventional way, we'll need to add a `Dataset` subclass like `_GLC_FCS30D_Dataset` to [`datasets.py`](https://github.com/aazuspan/sankee/blob/main/sankee/datasets.py) that implements the `get_year` method to retrieve the image for a requested year. You're probably already familiar with this, but I think we would need to grab the correct image based on the year range, then select the corresponding year band in that image.

Once that's done, we'd add a constant to the same file that instantiates that class with its metadata, labels, and palette, e.g. `GLC_FCS30D = _GLC_FCS30D_Dataset( ... )`, and finally add it to the `datasets` list at the bottom of the file.

Hopefully that all makes sense. You should be able to follow the template of other datasets in that file with a few changes (`get_year` will be the trickiest part). I'm happy to provide more guidance. Thank you!

**Comment by cchowgule** [comment]
@aazuspan, yes I see the issue with `_fetch_year_image`. Thanks for the pointers, I think I'm following you and if I get lost I will come back here with any questions.

The bigger issue, in my opinion, is the number of land-cover classes... The charts are going to be a mess. I created the table below based on [the paper published by the creators of the dataset](https://essd.copernicus.org/articles/16/1353/2024/). I chose the colors for the aggregated classes by just choosing the color of the first class within its group.

Should I include 1 dataset for each of the levels? Its not that much additional work to map classes into their aggregated groups.

The choice of colors for the aggregated groups feels pretty arbitrary. Do you have any suggestions for better ways of choosing a palette?

| Basic classification system | Level 1 validation system | Fine classification system |
|:----|:----|:----|
| Cropland `#ffff64` | Rainfed cropland `#ffff64` | Rainfed cropland `#ffff64` |
|    |    | Herbaceous cover cropland `#ffff64` |
|    |    | Tree or shrub cover cropland `#ffff00` |
|    | Irrigated cropland `#aaf0f0` | Irrigated cropland `#aaf0f0` |
| Forest `#4c7300` | Evergreen broadleaved forest `#4c7300` | Closed evergreen broadleaved forest `#4c7300` |
|    |    | Open evergreen broadleaved forest `#006400` |
|    | Deciduous broadleaved forest `#aac800` | Closed deciduous broadleaved forest `#aac800` |
|    |    | Open deciduous broadleaved forest `#00a000` |
|    | Evergreen needleleaved forest `#005000` | Closed evergreen needleleaved forest `#005000` |
|    |    | Open evergreen needleleaved forest `#003c00` |
|    | Deciduous needleleaved forest `#286400` | Closed deciduous needleleaved forest `#286400` |
|    |    | Open deciduous needleleaved forest `#285000` |
|    | Mixed-leaf forest `#a0b432` | Closed mixed-leaf forest `#a0b432` |
|    |    | Open mixed-leaf forest `#788200` |
| Shrubland `#966400` | Shrubland `#966400` | Shrubland `#966400` |
|    |    | Evergreen shrubland `#964b00` |
|    |    | Deciduous shrubland `#966400` |
| Grassland `#ffb432` | Grassland `#ffb432` | Grassland `#ffb432` |
| Tundra `#ffdcd2` | Lichens and mosses `#ffdcd2` | Lichens and mosses `#ffdcd2` |
| Bare areas `#ffebaf` | Bare areas `#fff5d` | Bare areas `#fff5d7` |
|    |    | Consolidated bare areas `#dcdcdc` |
|    |    | Unconsolidated bare areas `#fff5d7` |
|    | Sparse vegetation `#ffebaf` | Sparse vegetation `#ffebaf` |
|    |    | Sparse shrubland `#ffd278` |
|    |    | Sparse herbaceous cover `#ffebaf` |
| Wetland `#00a884` | Inland wetland `#00a884` | Swamp `#00a884` |
|    |    | Marsh `#73ffdf` |
|    |    | Flooded flat `#9ebbd7` |
|    |    | Saline `#828282` |
|    | Coastal wetland `#f57ab6` | Mangrove `#f57ab6` |
|    |    | Salt marsh `#66cdab` |
|    |    | Tidal flat `#444f89` |
| Impervious surface `#c31400` | Impervious surface `#c31400` | Impervious surface `#c31400` |
| Water body `#0046c8` | Water body `#0046c8` | Water body `#0046c8` |
| Permanent snow and ice `#ffffff` | Permanent snow and ice `#ffffff` | Permanent snow and ice `#ffffff` |

**Comment by aazuspan** [comment]
> I was curious though, my understanding of the Google Earth Engine is that any computations run on Google Cloud Compute and so they are very fast. Wouldn't it be more efficient to do the remapping on the cloud as opposed to using a DataFrame on the client machine?

It's hard to say for sure without knowing how GEE optimizes computations, but doing the remapping locally is just a much simpler task since we're remapping a few hundred rows rather than millions of pixels. I think the bigger advantage that I didn't mention there is that doing the remapping locally is also much easier to test and debug when it goes wrong.
